tasks/86_working-hours/working-hours.js

```diff
@@ -1,30 +1,71 @@
 /* eslint-disable max-len */
-export const formatWorkingHours = ( input ) => {
-  let days = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun', 'dummyDay'];
+// export const formatWorkingHours = ( input ) => {
+//   let days = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun', 'dummyDay'];
+//   if ( input.length ) {
+//     let result = [];
+//     let first = null; let last = null; let currentFrom = null; let currentTo = null;
+//     for ( let day of days ) {
+//       const workday = input.find((el) => el.day === day);
+
+//       if (first && ( !workday || currentFrom != workday.from || currentTo != workday.to )) {
+//         let firstLine = first.toUpperCase();
+//         if (last) firstLine += ` - ${last.toUpperCase()}`;
+//         firstLine += `: ${currentFrom} - ${currentTo}`;
+//         result.push( firstLine );
+//         first = null;
+//         last = null;
+//       }
+
+//       if ( workday && !first ) {
+//         first = workday.day;
+//         currentFrom = workday.from;
+//         currentTo = workday.to;
+//       } else if (first) {
+//         last = day;
+//       }
+//     }
+//     return result;
+//   }
+//   return [];
+// };
+
+//  Solution 2: It is really shitty code, with harcoded values but it's with grouping as Misha wanted me to do.
+//  I don't like this solution but at least I'm creative :-)  How to use reduce method in this case?
+
+let days = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'];
+export const formatWorkingHours = (input) =>{
   if ( input.length ) {
     let result = [];
-    let first = null; let last = null; let currentFrom = null; let currentTo = null;
-    for ( let day of days ) {
-      const workday = input.find((el) => el.day === day);
-
-      if (first && ( !workday || currentFrom != workday.from || currentTo != workday.to )) {
-        let firstLine = first.toUpperCase();
-        if (last) firstLine += ` - ${last.toUpperCase()}`;
-        firstLine += `: ${currentFrom} - ${currentTo}`;
-        result.push( firstLine );
-        first = null;
-        last = null;
-      }
+    for ( let value of Object.values(input)) {
+      result.push(Object.values(value));
+    }
+    let resultWithGroups = [];
+    let group1 = [];
+    let group2 = [];
+    let group3 = [];
+    let group4 = [];
 
-      if ( workday && !first ) {
-        first = workday.day;
-        currentFrom = workday.from;
-        currentTo = workday.to;
-      } else if (first) {
-        last = day;
+    for ( let i = 0; i < days.length; i++) {
+      let workday = result.find( (element) => element[0] === days[i]);
+      switch ( workday[1]) {
+        case '11:00':
```

Comment:
from my point of view it is not the best solution, that if workday will start at 9? at 8? at 7? it is not very good thing to relay to some values as they can change easily

Reply:
Agreed :)